File: CppPractice/CppLeetcode/Leetcode1684.cpp
#include <set>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    
    int countConsistentStrings(string allowed, vector<string>& words) 
    {
        set<char> goalSet;
        int ans = 0;
        for(char c:allowed)
        {
            goalSet.insert(c);
        }
        for(string word:words)
        {
            set<char> tempSet;
            bool isSubset = true;
            for(char c:word)
            {
                tempSet.insert(c);
            }
            for(char c:tempSet)
            {
                if(goalSet.find(c)==goalSet.end())
                {
                    isSubset = false;
                    break;
                }
            }
            if(isSubset)
            {
                ans++;
            }
        }
        return ans;
    }
};

File: JavaScriptPractice/Leetcode1684.js
var countConsistentStrings = function(allowed, words) 
{
    let ans=0;
    let goalSet  = new Set();
    for(const c of allowed)
    {
        goalSet.add(c);
    }
    for(const word of words)
    {
        let tempSet = new Set();
        let isSubset = true;
        for(const c of word)
        {
            tempSet.add(c)
        }
        for(const element of tempSet)
        {
            if(!(goalSet.has(element)))
            {
                isSubset = false;
                break;
            }
        }
        if (isSubset) {
            ans++;
        }
    }
    return ans;
};